### PR TITLE
VR Module refactoring improvement

### PR DESCRIPTION
**Pull request by KKarlash**
@bonewell 
@VladSemenyuk 
@DKlimenko 
@giangtqh 
@thinhts

### CODE REVIEW

**Comment by giangtqh** [comment]
please update CommandFactory to create appropriate command for ACTIVATE

File: src/components/vr_cooperation/src/command/base_json_request.cc [hunk 1]
```diff
@@ -90,6 +91,20 @@ vr_hmi_api::ResultCode BaseJsonRequest::GetHMIResultCode(
...
+    vr_hmi_api::RPCName function_id) {
+  switch (function_id) {
+    case vr_hmi_api::ON_REGISTER:
+      return functional_modules::REGISTER_SERVICE;
+    case vr_hmi_api::ACTIVATE:
+      return functional_modules::ACTIVATE_SERVICE;
+    case vr_hmi_api::ON_DEACTIVATED:
+      return functional_modules::ON_SERVICE_DEACTIVATED;
+    default:
+      return functional_modules::PROCESS_DATA;
```
Comment (bonewell) [inline src/components/vr_cooperation/src/command/base_json_request.cc]: I think in default case we should run plugin on wrong way. I mean it is better if the log has errors and warnings about sending unknown command.

File: src/components/vr_cooperation/src/command/activate_service_request.cc [hunk 2]
```diff
@@ -52,6 +52,11 @@ ActivateServiceRequest::~ActivateServiceRequest() {
 
 void ActivateServiceRequest::Execute() {
   LOG4CXX_AUTO_TRACE(logger_);
+  vr_hmi_api::ActivateServiceRequest params;
+  if (gpb_message().has_params() && params.ParseFromString(gpb_message().params())) {
+    int32_t app_id = params.appid();
+    parent()->ActivateService(app_id);
+   }
   application_manager::MessagePtr message_to_send;
   SendMessageToMobile(message_to_send);
```
Comment (bonewell) [inline src/components/vr_cooperation/src/command/activate_service_request.cc]: Is it OK that you send empty message?

**Comment by bonewell** [comment]
Reviewed.

File: src/components/vr_cooperation/src/command/activate_service_request.cc (hunk 2 above)
Comment (KKarlash) [inline src/components/vr_cooperation/src/command/activate_service_request.cc]: it will full fill in base class

Comment (bonewell) [inline src/components/vr_cooperation/src/command/activate_service_request.cc]: Ok, but I don't understand how base class knows about name and type of RPC.

File: src/components/vr_cooperation/src/command/base_json_request.cc (hunk 1 above)
Comment (DKlimenko) [inline src/components/vr_cooperation/src/command/base_json_request.cc]: Agree. Please add error msg and return error code

**Comment by VladSemenyuk** [comment]
Reviewed

**Comment by DKlimenko** [comment]
please check ini for socket and profile impl

File: src/components/vr_cooperation/src/socket_channel.cc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  DCHECK_OR_RETURN(socket_, false);
+  const size_t kChunkSize = 1000;
+  const size_t total_size = data.size();
+  size_t size_to_send = total_size < kChunkSize ? total_size : kChunkSize;
+  UInt8* data_bytes = new UInt8[size_to_send];
+
+  //TODO (KKarlash): add guard util
+  //utils::ScopeGuard ppsdata_guard = utils::MakeGuard(
+  //    utils::ArrayDeleter<UInt8*>, data_bytes);
+  //UNUSED(ppsdata_guard);
```
Comment (bonewell) [inline src/components/vr_cooperation/src/socket_channel.cc]: Who deletes data_bytes?

File: src/components/vr_cooperation/src/socket_channel.cc [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+bool SocketChannel::Receive(size_t size, std::string* buffer) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN(socket_, false);
+  const size_t kChunkSize = 1000;
+  size_t to_recv = size < kChunkSize ? size : kChunkSize;
+
+  size_t received = 0;
+  UInt8* data_bytes = new UInt8[size];
```
Comment (bonewell) [inline src/components/vr_cooperation/src/socket_channel.cc]: Who deletes data_bytes?

**Comment by DKlimenko** [comment]
reviewed

**Pull request merged**